evalml/data_checks/target_leakage_data_check.py

```diff
@@ -63,14 +62,17 @@ def validate(self, X, y):
         y = _convert_to_woodwork_structure(y)
         X = _convert_woodwork_types_wrapper(X.to_dataframe())
         y = _convert_woodwork_types_wrapper(y.to_series())
-
-        if y.dtype not in numeric_and_boolean_dtypes:
+        combined = X.copy()
+        combined['target'] = y
+        combined = _convert_to_woodwork_structure(combined)
+        mutual_info = combined.mutual_information()
```

Comment:
I think this is the easiest way to calculate mutual information, but it's also much more costly. Before, we were doing a O(n) implementation. Mutual information is O(n^2) which shouldn't be necessary here, given we only care about the correlation between the target and each column, and not each column with another. We should think about how to reduce this time complexity if we want to use mutual info!

Reply:
Great call! Will fix that.